src/api/app/controllers/webui/project_controller.rb

```diff
@@ -223,13 +223,22 @@ def restore
 
   def update
     authorize @project, :update?
+
+    if @project.update(project_params)
+      flash.now[:success] = 'Project was successfully updated.'
+    else
+      flash.now[:error] = 'Failed to update project'
+    end
+
     respond_to do |format|
-      if @project.update(project_params)
-        flash[:success] = 'Project was successfully updated.'
-      else
-        flash[:error] = 'Failed to update project'
+      format.html do
+        if request.xhr?
+          render partial: "edit_project", locals: { project: @project }, layout: false, status: :ok
+        else
+          redirect_to project_show_path(@project)
+        end
       end
-      format.html { redirect_to project_show_path(@project) }
+      format.js { render json: @project, status: :ok }
```

Comment:
we don't render json unless we use it to Datatables

Reply:
True, removing